src/query/sql/src/executor/physical_plan.rs

```diff
@@ -784,66 +789,41 @@ impl UnionAll {
 }
 
 #[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
-pub struct DistributedCopyIntoTableFromStage {
-    pub plan_id: u32,
+pub struct CopyIntoTable {
     pub catalog_info: CatalogInfo,
-    pub database_name: String,
-    pub table_name: String,
-    // ... into table(<columns>) ..  -> <columns>
     pub required_values_schema: DataSchemaRef,
-    // (1, ?, 'a', ?) -> (1, 'a')
     pub values_consts: Vec<Scalar>,
-    // (1, ?, 'a', ?) -> (?, ?)
     pub required_source_schema: DataSchemaRef,
-
     pub write_mode: CopyIntoTableMode,
     pub validation_mode: ValidationMode,
     pub force: bool,
-
     pub stage_table_info: StageTableInfo,
-    pub source: Box<DataSourcePlan>,
-
-    pub thresholds: BlockThresholds,
     pub files: Vec<StageFileInfo>,
     pub table_info: TableInfo,
+
+    pub source: CopyIntoTableSource,
 }
 
-impl DistributedCopyIntoTableFromStage {
-    pub fn output_schema(&self) -> Result<DataSchemaRef> {
-        Ok(DataSchemaRef::default())
-    }
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize, EnumAsInner)]
+pub enum CopyIntoTableSource {
+    Query(Box<QueryCtx>),
+    Stage(Box<DataSourcePlan>),
 }
 
 #[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
-pub struct CopyIntoTableFromQuery {
-    pub plan_id: u32,
-    pub catalog_info: CatalogInfo,
-    pub database_name: String,
-    pub table_name: String,
-
-    pub required_values_schema: DataSchemaRef, // ... into table(<columns>) ..  -> <columns>
-    pub values_consts: Vec<Scalar>,            // (1, ?, 'a', ?) -> (1, 'a')
-    pub required_source_schema: DataSchemaRef, // (1, ?, 'a', ?) -> (?, ?)
-    // these three fields are used for query result render
+pub struct QueryCtx {
```

Comment:
Please consider rename this struct,  too similar to the heavily use struct `QueryContext`...